src/ngx_lua_resty_lmdb_module.c

```diff
@@ -86,16 +113,30 @@ static char *
 ngx_lua_resty_lmdb_init_conf(ngx_cycle_t *cycle, void *conf)
 {
     ngx_lua_resty_lmdb_conf_t *lcf = conf;
+    const char*               cipher_type;
 
     ngx_conf_init_size_value(lcf->max_databases, 1);
     /* same as mdb.c DEFAULT_MAPSIZE */
     ngx_conf_init_size_value(lcf->map_size, 1048576);
+    cipher = (EVP_CIPHER *)EVP_aes_256_gcm();
+    if (lcf->encryption_type.data) {
+        cipher_type = (char *)lcf->encryption_type.data;
+        cipher = EVP_get_cipherbyname(cipher_type);
+        if (!cipher) {
```

Comment:
instead of just checking if `cipher` is not NULL (aka `cipher_type` is valid), we also need to make sure it's one of the two allowed ciphers (aes-256-gcm or chacha20-poly1305) and not others. using other ciphers especially non-AEAD ones will break our code.